ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/PostgresLedgerDao.scala

```diff
@@ -1073,15 +1074,15 @@ private class PostgresLedgerDao(
                     "package" -> p._1.toByteArray
                 )
               )
-            executeBatchSql(
-              SQL_INSERT_PACKAGE,
-              namedPackageParams
-            )
-            PersistenceResponse.Ok
-          }.recover {
-            case NonFatal(e) if e.getMessage.contains("duplicate key") =>
-              conn.rollback()
+            import scala.math.min
+            val changeCount = executeBatchSql(SQL_INSERT_PACKAGE, namedPackageParams)
+              .map(min(0, _))
+              .sum
+            if (changeCount == packages.length) {
+              PersistenceResponse.Ok
```

Comment:
I would suggest to change the return type of this to make sure the consumer can understand better. Ideally it would be `Map[PackageId, PersistenceResponse]` but I'm not sure it would be possible to get this information esaily. I guess `Map[PersistenceResponse, Int]` will do.